aspnetcore/fundamentals/target-aspnetcore.md

```diff
@@ -0,0 +1,213 @@
+
+## Use the ASP.NET Core shared framework
+
+With the release of .NET Core 3.0, many ASP.NET Core packages are no longer published to NuGet. Instead, the packages are included in the `Microsoft.AspNetCore.App` shared framework, which is installed with the .NET Core SDK. The shared framework was previously distributed as a NuGet package. For a list of packages no longer being published, see [Remove obsolete package references](xref:migration/22-to-30#remove-obsolete-package-references).
+
+As of .NET Core 3.0, projects using the `Microsoft.NET.Sdk.Web` MSBuild SDK implicitly reference the shared framework. Projects using the `Microsoft.NET.Sdk` or `Microsoft.NET.Sdk.Razor` SDK must target ASP.NET Core to use ASP.NET Core APIs in the shared framework.
+
+To target ASP.NET Core, add the following `<FrameworkReference>` element to your project file:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netcoreapp3.0-basic-library.csproj?highlight=8)]
+
+Targeting ASP.NET Core in this manner is only supported for projects targeting .NET Core 3.0.
+
+## Include UI components
+
+The following sections outline recommendations for libraries that include UI components. This guidance assumes the library won't multi-target. For guidance on supporting multiple ASP.NET Core versions, see [Support multiple ASP.NET Core versions](#support-multiple-aspnet-core-versions).
+
+### Razor views or Razor Pages
+
+A project that includes [Razor views](xref:mvc/views/overview) or [Razor Pages](xref:razor-pages/index) must use the [Microsoft.NET.Sdk.Razor SDK](xref:razor-pages/sdk).
+
+If the project targets .NET Core 3.0, it requires:
+
+* An `AddRazorSupportForMvc` MSBuild property set to `true`.
+* A `<FrameworkReference>` element for the shared framework.
+
+For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netcoreapp3.0-razor-views-pages-library.csproj)]
+
+If the project targets .NET Standard instead, it requires a [Microsoft.AspNetCore.Mvc](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc) package reference. The `Microsoft.AspNetCore.Mvc` package moved into the shared framework and is therefore no longer published. For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netstandard2.0-razor-views-pages-library.csproj?highlight=8)]
+
+### Razor components
+
+A project that includes [Razor components](xref:blazor/components) must use the [Microsoft.NET.Sdk.Razor SDK](xref:razor-pages/sdk).
+
+To support Razor component consumption from [Blazor Server](xref:blazor/hosting-models#blazor-server) only:
+
+* Target .NET Core 3.0.
+* Add a `<FrameworkReference>` element for the shared framework.
+
+For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netcoreapp3.0-razor-components-library.csproj)]
+
+To support Razor component consumption from both Blazor Server and [Blazor WebAssembly](xref:blazor/hosting-models#blazor-webassembly):
+
+* Target .NET Standard 2.0.
```

Comment:
I believe Blazor WebAssembly is moving to .NET Standard 2.1, but best to check timelines with @danroth27

Reply:
NS2.1 is current previews, so that's fine.

Reply:
Yeah, we moved to NET Standard 2.1 with the last Blazor WebAssembly preview.